File: server/server-core/src/main/java/uk/co/stevewellington/shoppinglist/actions/AddItemByNameToShoppingList.java
package uk.co.stevewellington.shoppinglist.actions;

import uk.co.stevewellington.shoppinglist.infrasctructure.ItemStore;
import uk.co.stevewellington.shoppinglist.model.ItemName;

import javax.inject.Inject;

public class AddItemByNameToShoppingList {
	
	private final ItemStore itemStore;
	
	@Inject public AddItemByNameToShoppingList(ItemStore itemStore) {
		this.itemStore = itemStore;
	}

	public void addItemByNameToShoppingList(ItemName itemName) {
		itemStore.createItemName(itemName);
		// TODO: something smells here. The business logic never touches the name inside ItemName
	}

}

File: server/server-core/src/main/scala/uk/co/stevewellington/shoppinglist/actions/AddItemByNameToShoppingList.scala
package uk.co.stevewellington.shoppinglist.actions

import com.google.inject.Inject
import uk.co.stevewellington.shoppinglist.infrasctructure.ItemStore
import uk.co.stevewellington.shoppinglist.model.ItemName

class AddItemByNameToShoppingList @Inject()(itemStore: ItemStore) {

  def addItemByNameToShoppingList(itemName: ItemName) = itemStore.createItemName(itemName)

}
